src/librustc_mir/borrow_check/nll/universal_regions.rs

```diff
@@ -479,6 +479,9 @@ impl<'cx, 'gcx, 'tcx> UniversalRegionsBuilder<'cx, 'gcx, 'tcx> {
         let mut indices = self.compute_indices(fr_static, defining_ty);
         debug!("build: indices={:?}", indices);
 
+        let _first_late_bound_index = self.infcx.num_region_vars();
+        self.infcx.replace_late_bound_regions_with_nll_infer_vars(self.mir_def_id, &mut indices);
```

Comment:
I think we only want to do this here *if* `self.mir_def_id != closure_base_def_id` -- i.e., if this is a closure/generator -- otherwise, we want to do it in the location you originally had it.